### ISSUE TITLE
get_metrics returns a result not consistent with PITLossWrapper

### ISSUE THREAD

**Issue by quancs** [asteroid-team/asteroid#509]
## 🐛 Bug

`get_metrics ` returns a result not consistent with `PITLossWrapper`

### To Reproduce

```
from asteroid.metrics import get_metrics
import torch
from asteroid.losses import PITLossWrapper,pairwise_neg_sisdr

while True:
    sources = torch.randn(1, 2, 64000)
    est_sources = torch.randn(1, 2, 64000)
    mixture = torch.randn(64000)


    m = get_metrics(mixture.numpy(), sources.numpy()[0, ], est_sources.numpy()[0, ], compute_permutation=True, metrics_list=['si_sdr'])
    print(m, "   compute_permutation=True")
    m = get_metrics(mixture.numpy(), sources.numpy()[0, ], est_sources.numpy()[0, ], compute_permutation=False, metrics_list=['si_sdr'])
    print(m, "   compute_permutation=False")

    asteroid_pit = PITLossWrapper(pairwise_neg_sisdr, pit_from='pw_mtx')
    loss_val = asteroid_pit(est_sources, sources)
    print(-loss_val.item(), "   PITLossWrapper(pairwise_neg_sisdr, pit_from='pw_mtx')")
    print()

```
### My Results

```
{'input_si_sdr': -64.04041290283203, 'si_sdr': -57.542795181274414}    compute_permutation=True
{'input_si_sdr': -64.04041290283203, 'si_sdr': -51.0589714050293}    compute_permutation=False 
-51.0589714050293    PITLossWrapper(pairwise_neg_sisdr, pit_from='pw_mtx')

{'input_si_sdr': -61.0158634185791, 'si_sdr': -61.65677261352539}    compute_permutation=True  
{'input_si_sdr': -61.0158634185791, 'si_sdr': -45.297006607055664}    compute_permutation=False
-45.29700469970703    PITLossWrapper(pairwise_neg_sisdr, pit_from='pw_mtx')

{'input_si_sdr': -47.6236515045166, 'si_sdr': -52.26988410949707}    compute_permutation=True 
{'input_si_sdr': -47.6236515045166, 'si_sdr': -62.26840019226074}    compute_permutation=False
-52.2698860168457    PITLossWrapper(pairwise_neg_sisdr, pit_from='pw_mtx')

{'input_si_sdr': -58.1720085144043, 'si_sdr': -57.48860168457031}    compute_permutation=True 
{'input_si_sdr': -58.1720085144043, 'si_sdr': -57.48860168457031}    compute_permutation=False
-50.58534240722656    PITLossWrapper(pairwise_neg_sisdr, pit_from='pw_mtx')

```

### Expected behavior

The `si_sdr` of `compute_permutation=True ` should be the same with `PITLossWrapper(pairwise_neg_sisdr, pit_from='pw_mtx')`. But, it's not.

The `si_sdr` of `compute_permutation=True ` should be higher than  `compute_permutation=False`. But, it's sometimes not.

#### Additional info

I don't know if it's my misunderstanding or not.

**Comment by quancs** [comment]
It seems that the problem is caused by `pb-bss-eval`

**Comment by mpariente** [comment]
Thanks for the bug report, it indeed looks like a bug, and thanks for the code. 

Would you have time to look into it deeper?

**Comment by quancs** [comment]
> 
> 
> Thanks for the bug report, it indeed looks like a bug, and thanks for the code.
> 
> Would you have time to look into it deeper?

I have implemented the si_sdr metric of "SDR – Half-baked or Well Done?". My results are the same with `PITLossWrapper(pairwise_neg_sisdr, pit_from='pw_mtx')`. So I guess the problem is  `pb-bss-eval`.

My implementation
```
def SISDR(target, estimate, EPS=1e-8):
    """ si_sdr of "SDR – Half-baked or Well Done?"
    """
    if target.shape != estimate.shape:
        raise TypeError(f"Inputs must be of shape [..., time], got {target.shape} and {estimate.shape} instead")

    mean_target = torch.mean(target, dim=-1, keepdim=True)
    mean_estimate = torch.mean(estimate, dim=-1, keepdim=True)
    target = target - mean_target
    estimate = estimate - mean_estimate

    α = torch.sum(estimate * target, dim=-1, keepdim=True) / (torch.sum(target**2, dim=-1, keepdim=True) + EPS)
    target_scaled = α * target

    noise = target_scaled - estimate

    si_sdr_value = torch.sum(target_scaled**2, dim=-1) / (torch.sum(noise**2, dim=-1) + EPS)
    si_sdr_value = 10 * torch.log10(si_sdr_value + EPS)

    return si_sdr_value
```

**Comment by mpariente** [comment]
Can you have a look [here](https://github.com/mpariente/pb_bss/blob/master/pb_bss_eval/evaluation/module_si_sdr.py) for differences please?

**Comment by quancs** [comment]
> 
> 
> Can you have a look [here](https://github.com/mpariente/pb_bss/blob/master/pb_bss_eval/evaluation/module_si_sdr.py) for differences please?

The results of [here](https://github.com/mpariente/pb_bss/blob/master/pb_bss_eval/evaluation/module_si_sdr.py) implementation are same  with mine. And, I have calculated the `input_si_sdr`, the results given by `pb-bss-eval` are the same with mine.  So, I guess the problem is in the PIT part of `pb-bss-eval`.

**Comment by mpariente** [comment]
That's what I expected, and this is a good news: the metrics is good, and the only problem is regarding the permutations. 

We usually don't compute it in the recipes, because we reorder before, but we should definitely fix it. 
Would you want to have a look? I'm writing my PhD these days, I don't have much time for that.

**Comment by quancs** [comment]
> 
> 
> That's what I expected, and this is a good news: the metrics is good, and the only problem is regarding the permutations.
> 
> We usually don't compute it in the recipes, because we reorder before, but we should definitely fix it.
> Would you want to have a look? I'm writing my PhD these days, I don't have much time for that.

OKay, I will try, but I cannot guarantee it will be fixed soon. I'm writing my dissertation proposal...

**Comment by quancs** [comment]
`pb_bss_eval.OutputMetrics` uses the `perm` returned from `pb_bss_eval.evaluation.mir_eval_sources` from `mir_eval.separation.bss_eval_sources`. That's the problem. 

And, I checked the `perm` returned from `mir_eval.separation.bss_eval_sources`: it's not the same with the `perm` returned from my PIT implementation, while my PIT returns the same results as `PITLossWrapper`.

**Comment by mpariente** [comment]
This does make sense. 
Is it a batching problem maybe? Or just that the optimal permutation for SDR is not the same?
Do you have an idea how to fix it?

**Comment by quancs** [comment]
> 
> 
> > Maybe it is just that the optimal permutation for SDR is not the same with the optimal permutation for SI-SDR .
> 
> I thought abut that, but I'm not convinced.

Yes, it is my guess. I could not figure it out, because the sdr in `bss_eval_sources` is a little bit complex, and I did not implement it yet. I mean I could not figure out if it is a bug of `bss_eval_sources` or if it is the reason described above.

> Maybe we should just disable it for now, until we solve it, because PIT can be used instead anyway, and it's reliable.

I think this is a good choice.

**Comment by mpariente** [comment]
I don't think there is a bug there, it's reliable. 
Do you want to submit a PR to raise an error for compute_permutation=True? That would be cool.

**Comment by quancs** [comment]
> 
> 
> I don't think there is a bug there, it's reliable.
> Do you want to submit a PR to raise an error for compute_permutation=True? That would be cool.

Thank you. You can fix it.
